src/daft-local-execution/src/intermediate_ops/project.rs

```diff
@@ -31,10 +37,13 @@ impl IntermediateOperator for ProjectOperator {
         input: Arc<MicroPartition>,
         state: Box<dyn IntermediateOpState>,
         runtime: &RuntimeRef,
+        memory_manager: Arc<MemoryManager>,
     ) -> IntermediateOpExecuteResult {
         let projection = self.projection.clone();
+        let memory_request = self.memory_request;
         runtime
             .spawn(async move {
+                let _permit = memory_manager.request_bytes(memory_request).await?;
```

Comment:
IIRC this is only used for UDFs today, and other operations (such as scanning Parquet) will not be requesting for memory.

Is the intention to extend this functionality to also performing memory requests for other operators?

Reply:
Yeah it’s just udfs for now, and i intend to extend this to other operators